File: HW7 submission/Leibniz.py

# Programming for Engineers - Homework 7
# Elana Lapins, due: 11/24/2021

import math

cases = int(input())

for i in range(cases):
    print("Case: ", i, "\n")

    n = int(input())
    pifour =0

    for i in range(0,n):
        pifour += ( (-1)**i)/( (2*i)+1)

    Pi = pifour*4;

    print("Pi is estimated as: {:.4f}".format(Pi))

    

File: Liebniz_Flip_Money_Cpp/LEIB/leibniz.cpp
// Elana Lapins - Homework 2 Problem 3: Leibniz
// Programming for Engineers, due: 09/17/2021

#include <iostream>
#include<cmath>
#include<iomanip>

int main() {
	int cases = 0;
	std::cin >> cases;
	//double Pi = 0;
	//double pifour = 0;

	// Leibniz series: pi/4 = sum(n=0 -> inf) { 1 - (-1)^n / (2n+3) }

	for (int i = 0; i < cases; i++) {
		double n = 0;
		std::cout << "Case " << i << ":\n";
		std::cin >> n;
		double Pi = 0;
		double pifour = 0;

		for (double N = 0; N <= n - 1; N++) {
			pifour += (pow(-1, N)) / (2 * N + 1);
		}
		Pi = pifour * 4;
		std::cout << "Pi estimated as: " << std::fixed << std::setprecision(8) << Pi << "\n";
	}
}